cmd/kube-proxy/app/server_others.go

```diff
@@ -164,6 +164,56 @@ func newProxyServer(
 			healthzServer,
 			config.NodePortAddresses,
 		)
+
```

Comment:
The func can be simplified by moving all the creation code into private funcs. So eventually it looks like this:
```
if <validation - step0..stepN>{
// failed return error
}

// if dualstack set dualstackFlag true
// if ipvs set ipvs 
}
  if ipvs{
     createIPVs(.. all args, dualstackFlag)
     return
  }

createIPTables(..all args, dualstackflag)
return
```
`createIPVs` and `createIPTables` are new funcs that only do either dualstack or not.

Reply:
it should allow us to create a unit test to cover all configuration possibilities and the return value of the creation func.